analysis/variables_cohorts.py

```diff
@@ -0,0 +1,584 @@
+        .first_for_patient()
+        .appointment_date
+    )
+
+    tmp_exp_covid19_confirmed_death = matching_death_between(covid_codes, index_date, end_date_exp)
+
+    tmp_exp_date_death = ons_deaths.date
+
+    tmp_exp_date_covid19_confirmed_death = case(
+        when(tmp_exp_covid19_confirmed_death).then(tmp_exp_date_death)
+    )
+
+    ## Define individual temporary variables (subgroup variables) first before using them in the dictrionary
+
+    ## 1. Covid-19 prior to study start date (2020-01-01)
+
+    ### SGSS
+
+    tmp_sub_bin_priorcovid19_confirmed_sgss = (
+        sgss_covid_all_tests.where(
+            sgss_covid_all_tests.specimen_taken_date.is_before(index_date)
+        )
+        .where(sgss_covid_all_tests.is_positive)
+        .exists_for_patient()
+    )
+
+    ### Primary care
+
+    tmp_sub_bin_priorcovid19_confirmed_snomed = (
+        clinical_events.where(
+            (clinical_events.ctv3_code.is_in(
+                covid_primary_care_code + 
+                covid_primary_care_positive_test + 
+                covid_primary_care_sequalae)) &
+            clinical_events.date.is_before(index_date)
+        )
+        .exists_for_patient()
+    )
+
+    ### SUS
+
+    tmp_sub_bin_priorcovid19_confirmed_apc = (
+        apcs.where(
+            ((apcs.primary_diagnosis.is_in(covid_codes)) | (apcs.secondary_diagnosis.is_in(covid_codes))) & 
+            (apcs.admission_date.is_before(index_date))
+        )
+        .exists_for_patient()
+    )
+
+    tmp_sub_bin_priorcovid19_confirmed_opa = (
+        opa_diag.where(
+            ((opa_diag.primary_diagnosis_code.is_in(covid_codes)) | (opa_diag.secondary_diagnosis_code_1.is_in(covid_codes))) & 
+            (opa_diag.appointment_date.is_before(index_date))
+        )
+        .exists_for_patient()
+    )
+
+    ## 2. Covid-19 severity
+
+    ### SUS
+
+    tmp_sub_date_severecovid19_apc = (
+        apcs.where(
+            ((apcs.primary_diagnosis.is_in(covid_codes)) | (apcs.secondary_diagnosis.is_in(covid_codes))) & 
+            (apcs.admission_date.is_on_or_after(index_date))
+        )
+        .sort_by(apcs.admission_date)
+        .first_for_patient()
+        .admission_date
+    )
+
+    tmp_sub_date_severecovid19_opa = (
+        opa_diag.where(
+            ((opa_diag.primary_diagnosis_code.is_in(covid_codes)) | (opa_diag.secondary_diagnosis_code_1.is_in(covid_codes))) & 
+            (opa_diag.appointment_date.is_on_or_after(index_date))
+        )
+        .sort_by(opa_diag.appointment_date)
+        .first_for_patient()
+        .appointment_date
+    )
+
+    ## prepare for smoking status
+
+    tmp_most_recent_smoking_cat = (
+        last_matching_event_clinical_ctv3_before(smoking_clear, index_date)
+        .ctv3_code.to_category(smoking_clear)
+        )
+
+    tmp_ever_smoked = ever_matching_event_clinical_ctv3_before(ever_current_smoke, index_date)   # uses a different codelist with ONLY smoking codes
```

Comment:
What does this comment mean by 'ONLY smoking codes'?

Reply:
https://github.com/opensafely/early-inflammatory-arthritis/blob/069e61712fcc9a0c2ec2804ff36a9b773073291c/analysis/dataset_definition.py#L149-L160u

useful codes from @milanwiedemann

Reply:
Okay - do we know how this differs from what we were doing in terms of how many people change classification? I am conscious that we are trying to replicate at the moment so want to be making minimal number of changes.

Reply:
I think they are the equivalent codes to those used in previous Cohort Extractor projects:
https://github.com/opensafely/post-covid-mentalhealth/blob/ddb50bd8fc9e4e1a9855dc9f9a03991487c908aa/analysis/common_variables.py#L617-L630

Reply:
Sorry, Zoe - I don't follow - if the code lists are identical, why did you swap them? And how many people change classification as a result of correcting the "ever smoke" category?

Reply:
Apologies for the confusion. The functionality remains identical, but the previous code required manually extracting the codelist for "ever smoke." In the updated version, I replaced it with the `filter_codes_by_category()` command, which I learned from Milan's code. This makes codes shorter.